app/src/androidTest/java/com/alexzh/testapp/ui/home/HomeActivityTest.kt

```diff
@@ -18,77 +18,95 @@ import org.junit.runner.RunWith
 @RunWith(AndroidJUnit4::class)
 class HomeActivityTest {
 
-    @Rule
-    @JvmField
+    /**
+     * The {@link ActivityTestRule} provides functional testing of a single {@link Activity}.
+     * The Espresso framework requires it for launching Activity for testings.
+     *
+     * We can configure activity before running in the following ways:
+     *  - Using the {@link #ActivityTestRule(Class, boolean, boolean)} where the second parameter
+     *  is initialTouchMode and the third one is launchActivity. If launch activity is true the
+     *  Activity will be launched before the test case. It means we cannot configure it before
+     *  executing the test case. If we want to configure which intent launch activity and pass
+     *  additional parameters we can execute {@see activity#launchActivity(Intent)} method with
+     *  required intent.
+     *  - Using the {@link ActivityTestRule#beforeActivityLaunched) method we can do additional
+     *  action before executing test case, like configure Mock objects, clean database, etc.)
```

Comment:
Maybe it should be mentioned that the `beforeActivityLaunched` needs to be overriden in the rule itself like:

```java
@Rule
public ActivityTestRule<FooActivity> testRule = new ActivityTestRule<FooActivity>(FooActivity.class) {
    @Override
    protected void beforeActivityLaunched() {
        super.beforeActivityLaunched();
        // custom code here?
    }
};
```

Maybe this should be a full example?

Reply:
Maybe this could be a whole section of the workshop. Here you have only short explanations. Then in some exercise you could show different ways of handling the ActivityTestRule? This seems like a nice material for a few exercises.

Reply:
I want to cover it with slides. That's a great idea to describe it with example

Reply:
done